src/server/service/debug.rs

```diff
@@ -38,30 +41,78 @@ fn error_to_grpc_error(tag: &'static str, e: Error) -> GrpcError {
     e
 }
 
+pub type Callback<T> = Box<dyn FnOnce(T) + Send>;
+pub type ResolvedTsDiagnosisCallback = Callback<(
+    bool, // exist
+    bool, // stopped
+    u64,  // resolved_ts
+    u64,  // tracked index
+    u64,  // num_locks
+    u64,  // num_transactions
+)>;
+pub type ScheduleResolvedTsTask = Arc<
+    dyn Fn(
+            u64,  // region id
+            bool, // log_locks
+            u64,  // min_start_ts
+            ResolvedTsDiagnosisCallback,
+        ) -> bool
+        + Send
+        + Sync,
+>;
+
 /// Service handles the RPC messages for the `Debug` service.
-#[derive(Clone)]
-pub struct Service<T, D>
+pub struct Service<T, D, S>
 where
-    T: RaftExtension,
-    D: Debugger,
+    T: RaftExtension + Clone,
+    D: Debugger + Clone,
+    S: StoreRegionMeta,
 {
     pool: Handle,
     debugger: D,
     raft_router: T,
+    store_meta: Arc<Mutex<S>>,
+    resolved_ts_scheduler: ScheduleResolvedTsTask,
```

Comment:
```suggestion
    resolved_ts_scheduler: Arc<Option<Scheduler<Task>>>,
```

What about pass the scheduler so that it'll be easier to schedule other tasks if needed.

Reply:
Resolved-ts depends on tikv, so we cannot make tikv depend on resolved-ts. That's why I used this ugly way to pass the scheduler..